pkg/ebpf/bpf_error_decoder.go

```diff
@@ -0,0 +1,86 @@
+package ebpf
+
+import (
+	"encoding/binary"
+	"fmt"
+	"unsafe"
+)
+
+type bufferDecoder struct {
+	cursor       int
+	delimiter    int
+	bufferLength int
+}
+
+func (d *bufferDecoder) updateCursor(increase int) {
+	d.cursor += increase
+}
+
+func (d *bufferDecoder) isCursorOutOfBounds() bool {
+	return d.cursor >= d.bufferLength
+}
+
+func (d *bufferDecoder) updateDelimiter(increase int) error {
+	return d.setDelimiter(d.cursor + increase)
+}
+
+func (d *bufferDecoder) setDelimiter(newDelimiter int) error {
+	if newDelimiter > d.bufferLength {
+		return fmt.Errorf("decode: delimiter %d out of bounds %d", newDelimiter, d.bufferLength-1)
+	}
+	d.delimiter = newDelimiter
+
+	return nil
+}
+
+// decodeBuffer decodes rawBuffer updating fields
+// - Fields must be passed as pointers
+// - The fields order matters
+func decodeBuffer(rawBuffer []byte, fields ...interface{}) error {
```

Comment:
Ok, so this is a bit similar to how the `binary.Read` method works minus the reflection option, which we've used in the past but moved from.
We don't really need the switch case type versatility here since we know ahead of time what our field and type order is in the buffer, so we should do the same here as we do in bufferdecoder in that regard, without passing in `fields ...interface{}`, it's unnecessary flexibility.

Reply:
I simplified the logic in bpf_error.go, getting rid of the outer (file) decoder. The decoder test was removed as well, as it's trivial now.